### PR TITLE
feat(NFDIV-530): null unreachable answers

### PR DESCRIPTION
**Pull request by SebFlippence**
### JIRA link ###

[NFDIV-530](https://tools.hmcts.net/jira/browse/NFDIV-530) Clear previously entered, but now unreachable answers

### Change description ###

Sets previously entered, but now unreachable answers to `null`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by velimd — APPROVED** [review]

File: src/main/app/controller/PostController.ts [hunk 1]
```diff
@@ -63,6 +69,20 @@ export class PostController<T extends AnyObject> {
       res.redirect(nextUrl);
     });
   }
+
+  private getAnsweredUnreachableResponsesAsNull(req: AppRequest<T>): Partial<Case> {
```
Comment (linusnorton) [inline src/main/app/controller/PostController.ts:73]: stop shoving everything in the post controller :D

**Review by SarahLittlejohn — APPROVED** [review]

**Review by linusnorton — CHANGES_REQUESTED** [review]
I think setting an answer should knock out other fields

**Comment by velimd** [comment]
> I think setting an answer should knock out other fields

That would be a pain for a user who is almost done but they change something which might not effect any path such as date on certificate or date of birth, which will make them start all over again. At that point, i rather stay married.

**Comment by linusnorton** [comment]
> > I think setting an answer should knock out other fields
> 
> At that point, i rather stay married.

All part of the plan

File: src/main/app/controller/PostController.ts (hunk 1 above)
Comment (SebFlippence) [reply src/main/app/controller/PostController.ts]: 👍 Fixed in 07670ab.

**Review by linusnorton — APPROVED** [review]

**Pull request merged**